src/main/java/net/minecraftforge/event/datafix/ConfigureDataFixSchemaEvent.java

```diff
@@ -0,0 +1,208 @@
+    {
+        registerBlockEntity(registryName, () -> template.apply(registryName.toString()));
+    }
+
+    /**
+     * Allows for the custom definition of a new block entity type and registration of it to the current schema and all its descendent versions.
+     * The supplier is invoked to produce the block entity type template.
+     *
+     * @param registryName The registry name of the block entity type.
+     * @param template The supplier of the type template.
+     */
+    public void registerBlockEntity(final ResourceLocation registryName, final Supplier<TypeTemplate> template)
+    {
+        blockEntityTypes.put(registryName.toString(), template);
+    }
+
+    /**
+     * Removes an already registered block entity from the current schema and all its descendent versions.
+     * This allows for the removal of a custom block entity type when it is registered in a parent version and is no longer needed.
+     *
+     * @param registryName The registry name of the block entity to remove.
+     */
+    public void removeBlockEntity(final ResourceLocation registryName)
+    {
+        blockEntityTypes.put(registryName.toString(), null);
+    }
+
+    /**
+     * Registers a new simple (catch-all-nbt) entity to the current schema and all its descendent versions.
+     *
+     * @param registryName The registry name of the entity.
+     */
+    public void registerSimpleEntity(final ResourceLocation registryName)
+    {
+        registerEntity(registryName, DSL::remainder);
+    }
+
+    /**
+     * Allows for the custom definition of a new entity type and registration of it to the current schema and all its descendent versions.
+     * The function is given the registry name of the entity type template to produce.
+     *
+     * @param registryName The registry name of the entity type.
+     * @param template The function to produce the entity type template.
+     */
+    public void registerEntity(final ResourceLocation registryName, final Function<String, TypeTemplate> template)
+    {
+        registerEntity(registryName, () -> template.apply(registryName.toString()));
+    }
+
+    /**
+     * Allows for the custom definition of a new entity type and registration of it to the current schema and all its descendent versions.
+     * The supplier is invoked to produce the entity type template.
+     *
+     * @param registryName The registry name of the entity type.
+     * @param template The supplier of the type template.
+     */
+    public void registerEntity(final ResourceLocation registryName, final Supplier<TypeTemplate> template)
+    {
+        entityTypes.put(registryName.toString(), template);
+    }
+
+    /**
+     * Removes an already registered entity from the current schema and all its descendent versions.
+     * This allows for the removal of a custom entity type when it is registered in a parent version and is no longer needed.
+     * 
+     * @param registryName The registry name of the entity to remove.
```

Comment:
Should this be entity type instead of just entity?

Reply:
No this does not refer to the type, but to the NBT of the actual entity.